crates/router/src/connector/braintree/braintree_graphql_transformers.rs

```diff
@@ -56,21 +84,29 @@ pub struct TransactionBody {
     merchant_account_id: Secret<String>,
 }
 
-impl TryFrom<&types::PaymentsAuthorizeRouterData> for BraintreePaymentsRequest {
+impl TryFrom<&BriantreeAmountConversion<&types::PaymentsAuthorizeRouterData>>
+    for BraintreePaymentsRequest
+{
     type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+    fn try_from(
+        item: &BriantreeAmountConversion<&types::PaymentsAuthorizeRouterData>,
+    ) -> Result<Self, Self::Error> {
         let metadata: BraintreeMeta =
-            utils::to_connector_meta_from_secret(item.connector_meta_data.clone())?;
-        utils::validate_currency(item.request.currency, metadata.merchant_config_currency)?;
+            utils::to_connector_meta_from_secret(item.req.connector_meta_data.clone())?;
+        utils::validate_currency(item.req.request.currency, metadata.merchant_config_currency)?;
 
-        match item.request.payment_method_data.clone() {
+        match item.req.request.payment_method_data.clone() {
             api::PaymentMethodData::Card(_) => {
-                if item.is_three_ds() {
+                if item.req.is_three_ds() {
                     Ok(Self::CardThreeDs(BraintreeClientTokenRequest::try_from(
                         metadata,
                     )?))
                 } else {
-                    Ok(Self::Card(CardPaymentRequest::try_from((item, metadata))?))
+                    Ok(Self::Card(CardPaymentRequest::try_from((
+                        item.req,
+                        metadata,
+                        &item.amount,
```

Comment:
Change try_from implementation to (&BriantreeAmountConversion<&types::PaymentsAuthorizeRouterData>, metadata)